src/cypress/support/commands.js

```diff
@@ -0,0 +1,133 @@
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+let currentPrice = ''
+let textList
+let notional1
+let finalList = []
+let finalAmount = 0
+const dayjs = require('dayjs')
+var currentYear = dayjs().get('year')
+let currency
+
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+
+Cypress.Commands.add('VerifyCurrencyOnPage', symbol => {
+  let currency = symbol.replace('/', '')
+
+  cy.log('').then(() => {
+    cy.get('div[data-testid="tile-' + currency + '"]')
+      .scrollIntoView()
+      .should('be.visible')
+  })
+})
+
+Cypress.Commands.add('performTrade', (symbol, notional, direction) => {
+  currency = symbol.replace('/', '')
+  notional1 = notional
+
+  cy.log('').then(() => {
+    cy.get(
+      'div[data-testid="tile-' +
+        currency +
+        '"] > div > div > div:nth-child(2) div:nth-child(2) input'
+    )
+      .clear()
+      .type(notional)
+    cy.get(
+      'div[data-testid="tile-' +
+        currency +
+        '"] > div > div > div:nth-child(2)  > div:nth-child(2) > div > button[direction=' +
+        direction +
+        '] > div > div'
+    ).each((el, index, list) => {
+      if (index == 0) {
+        currentPrice = el.find('div:nth-child(2)').text()
+      } else {
+        currentPrice = currentPrice + el.text()
+      }
+    })
+  })
+
+  cy.log('').then(() => {
+    cy.log('current price is ' + currentPrice)
+    finalAmount = Number(currentPrice) * Number(notional)
+
+    cy.get(
+      'div[data-testid="tile-' +
+        currency +
+        '"] > div > div > div:nth-child(2)  > div:nth-child(2) > div > button[direction=' +
+        direction +
+        ']'
+    ).click({ force: true })
+    cy.wait(5000)
+    cy.get('div[data-testid=tile-' + currency + '] div[role=dialog] > div:nth-child(2)').then(
+      el => {
+        let tradeIDList = el.text().split(' ')
+        finalList.push(tradeIDList[2])
+      }
+    )
+    cy.get('div[data-testid=tile-' + currency + '] div[role=dialog] > div[role=alert]').then(el => {
+      cy.log('Confirmation text ' + el.text())
+
+      textList = el.text().split(' ')
+      finalList.push('Done')
+      finalList.push(textList[14] + '-' + textList[15].replace('.', '') + '-' + currentYear)
+      finalList.push(direction)
+      finalList.push(textList[2] + textList[10])
+      if (direction.toLowerCase() == 'buy') {
+        finalList.push(textList[2])
+      } else {
+        finalList.push(textList[10])
+      }
+      finalList.push(textList[3])
+      finalList.push(textList[8])
+      finalList.push(textList[14] + '-' + textList[15].replace('.', '') + '-' + currentYear)
+      finalList.push('JPW')
+    })
+  })
+})
+
+Cypress.Commands.add('verifyTradeSuccess', () => {
+  cy.log('').then(() => {
+    expect(Number(currentPrice)).to.eq(Number(finalList[7]))
```

Comment:
I mentioned before, but the concept of extracting bits into some `finalList` state is confusing and hard to reason about.
To verify a trade success I think we can be a bit simpler by just asserting that the tile contains a string that we expect.
Take a look at my early examples I originally showed you - https://github.com/taylor-adaptive/ReactiveTraderCloud/blob/feat/cyrpress/src/new-client/cypress/integration/trade-execution.ts#L28
I find this a lot easier to reason about, and therefor maintain. Interested to hear your thoughts.

Finally, for the bits that we can't really do nicely (like asserting on the current price) I think it's acceptable to leave it as a TODO until we have a mocking solution (similar to what I did in my earlier examples)

Reply:
I am using finalList to make it scalable, Lets if in future we need to integrate blotter tests, where after trade we need to check details in blotter row. then finaList will help us in fast assertions. 

we can give any other name to it, if it sounds confusing.

Reply:
I would say the opposite, I think using an array to hold random values is not scalable.

For example - 
There is no way of knowing what `finalList[6]` holds unless you step through the whole code and count through the pushes to `finalList`, and then that contains a `textList[x]` which we then have to look at what `textList` is and try and figure out what value that holds. As someone who may need to read and update these tests, it will be difficult.

I'm not an expert with testing so I am open to solutions, but this doesn't feel like the best way to go about things.

@lidia-adaptive can you take a look through this code and the early examples I mention in the above comment and offer your opinion on approaches?

Reply:
Absolutely agree with you @taylor-adaptive, this is a mystery-guest or magic number anti-pattern right there. 
Looking at your example.